src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax;
+
+public abstract class Figure {
+    private Color color;
+    private double area;
+
+    public Figure(Color color) {
+        this.color = color;
+    }
+
+    public void setArea(double area) {
+        this.area = area;
+    }
+
+    @Override
+    public String toString() {
+        return " color = " + color
+                + ", area = " + area + '.';
+    }
+
+    public abstract double areaCalculator();
```

Comment:
lets create interfaces some `AreaCalculator` and `InfoDrawer` with two abstract methods `calculateArea()` and `draw()` and implement them in `Figure` class